base/ltcmd.dtx

```diff
@@ -3873,6 +3954,166 @@
 %    \end{macrocode}
 % \end{macro}
 %
+% \subsection{Conversion to key--value form}
+%
+% This is implemented as a process but with no public interfaces,
+% hence is treated separately from the others: it's a feature of
+% \pkg{ltcmd} which just happens to use the same mechanism as a processor.
+%
+% \begin{macro}{\@@_arg_to_keyvalue:nn}
+% \changes{v1.1a}{2022/08/10}{New internal arg-to-keyval processor}
+% \begin{macro}{\@@_arg_to_keyvalue_auxi:nnn}
+% \begin{macro}{\@@_arg_to_keyvalue_auxii:nnNw}
+% \begin{macro}{\@@_arg_to_keyvalue_auxiii:nnn}
+% \begin{macro}{\@@_arg_to_keyvalue_auxiv:nnNw}
+% \begin{macro}{\@@_arg_to_keyvalue_auxv:nn}
+% \begin{macro}{\@@_arg_to_keyvalue_loop:nnw}
+% \begin{macro}{\@@_arg_to_keyvalue_loop_group:nnn}
+% \begin{macro}{\@@_arg_to_keyvalue_loop_space:nnw}
+% \begin{macro}{\@@_arg_to_keyvalue_loop_N_type:nnN}
+% \begin{macro}{\@@_arg_to_keyvalue_math:nnw}
+% \begin{macro}{\@@_arg_to_keyvalue_math_N_type:nnN}
+% \begin{macro}{\@@_arg_to_keyvalue_math_group:nnn}
+% \begin{macro}{\@@_arg_to_keyvalue_math_space:nnw}
+%   If the entire argument is braced, we treat as free text and return as
+%   the value for the text key. Alternatively, if the start of the input is
+%   |=,| then it is forced to be key--value. To avoid needing to worry about
+%   catcodes for this, and to allow spaces around the |=|, we use a
+%   series of steps rather than a delimited argument.
+%     \begin{macrocode}
+\cs_new_protected:Npn \@@_arg_to_keyvalue:nn #1#2
+  {
+    \str_if_eq:eeTF { \exp_not:n {#2} } { \exp_not:o { { \use:n #2 } } }
```

Comment:
The `\exp_not:o` here does nothing useful.  I added a `\exp_after:wN` here, but the test output for:
```
    \DeclareDocumentCommand \foo { = { TOC-entry } m } { \TYPE { \detokenize { (#1) } } }
    \foo { { baz } }
```
changes from `(TOC-entry={{baz}})` to `(TOC-entry={baz})`.  I'm not sure which is correct...

Reply:
If I understand correctly, this is testing if the grabbed argument is wrapped in a set of braces.  If so, this fails for the (likely unsupported) case of different bgroup and egroup characters.  For example:
```
\DeclareDocumentCommand \foo { = { TOC-entry } m } { \typeout { \detokenize { (#1) } } }
\foo { { baz } }
\catcode`\[=1
\catcode`\]=2
\foo [ [ baz ] ]
```
will print:
```
(TOC-entry={baz})
(TOC-entry={[baz]})
```

A version that protects against that would use (I think) `\tl_trim_spaces:n`, then `\tl_if_head_is_group:nTF`, and then test `\tl_if_blank:oTF { \use_none:n #2 }`.  Something like this:
```
\cs_new_protected:Npn \@@_arg_to_keyvalue:nn #1#2
  { \tl_trim_spaces_apply:nN {#2} \@@_arg_to_keyvalue_braces:nn {#1} }
\cs_new_protected:Npn \@@_arg_to_keyvalue_braces:nn #1 #2
  {
    \tl_if_head_is_group:nT {#1}
      {
        \tl_if_blank:oT { \use_none:n #1 }
          {
            \tl_set:Nn \ProcessedArgument { #2 = #1 }
            \use_none:nn
          }
      }
    \use:n
      {
        \exp_args:Ne \@@_arg_to_keyvalue_auxi:nnn
          { \tl_trim_spaces:n {#1} } {#2} {#1}
      }
  }
```

Reply:
I've made the change also normalising the brace group